src/gui/qgsmapcanvas.h

```diff
@@ -1034,11 +1034,21 @@ class GUI_EXPORT QgsMapCanvas : public QGraphicsView
 
     /**
      * Emitted when the map canvas temporal range changes.
-    *
-    * \since QGIS 3.14
-    */
+     *
+     * \since QGIS 3.14
+     */
     void temporalRangeChanged();
 
+
+    /**
+     * Emitted after the map canvas context menu has been populated.
+     * Can be used to extend the map canvas context menu before it will be shown.
+     *
+     * \since QGIS 3.16
+     */
```

Comment:
I think a more descriptive name would be "contextMenuAboutToShow" . The use of "populated" makes it seem that it's already too late to manipulate the menu contents.